lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -503,6 +504,14 @@ def add_marketplace_data(post, options)
         end
       end
 
+      def add_amount_allocations_data(post, options)
+        return unless options[:amount_allocations]&.is_a?(Array)
```

Comment:
nit: it doesn't hurt but you don't really need the null safe here, `nil.is_a?(Array)` should work just fine.